mage/k3d/lib.go

```diff
@@ -103,9 +104,28 @@ func reloadLocalImage(image string) {
 }
 
 func setupKubeconfig() {
-	config := shutil.OutputPanic("k3d", "get-kubeconfig")
-	os.Setenv("KUBECONFIG", config)
-	fmt.Printf("To set up kubectl in your shell to use this cluster, set:\n\tKUBECONFIG=%s\n", config)
+	k3dConfig := shutil.OutputPanic("k3d", "get-kubeconfig")
+	currentConfig := kubectl.GetKubeconfig()
+
+	// merge current config + new k3d config
+	// the new config must appear before the current config to update it
+	// when merging
+	os.Setenv("KUBECONFIG", fmt.Sprintf("%s:%s", k3dConfig, currentConfig))
+	merged := shutil.OutputPanic("kubectl", "config", "view", "--flatten")
+
+	// look up current config file so we can retain its permissions
+	// before overwriting
+	configInfo, err := os.Stat(currentConfig)
```

Comment:
this could technically error out if your kubeconfig wasn't specified and `~/.kube/` dir did not contain a `config` file... but i'm not sure if its worth adding logic to create an empty default in that case (is that even a real concern?)

Reply:
I'm sure someone will hit that, we can just document somewhere they should `mkdir -p ~/.kube && touch ~/.kube/config`